server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java

```diff
@@ -38,6 +38,21 @@
 
 public class JavaJodaTimeDuellingTests extends ESTestCase {
 
+    //these parsers should allow both ',' and '.' as a decimal point
+    public void testDecimalPointParsing(){
+        assertSameDate("2001-01-01T00:00:00.123Z", "strict_date_optional_time");
```

Comment:
I think parsing of a date like `2001-01-01T00:00:00.456,123Z` should not work because the nanos have already been parsed. Just want to be sure here.

Reply:
fair point, the fix allows that atm

Reply:
are you sure? I havent tested this, but this might throw an exception because of parsing the same type twice.

Reply:
from the test I tried it was not failing. 
sadly java.time does not have the append method on a builder with multiple parsers
https://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormatterBuilder.html#append-org.joda.time.format.DateTimePrinter-org.joda.time.format.DateTimeParser:A-

https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatterBuilder.html#append-java.time.format.DateTimeFormatter-

Reply:
actually it fails when the value is different, like in your example 
`2001-01-01T00:00:00.456,123Z`
but will pass for
`2001-01-01T00:00:00.123,123Z`

Reply:
just to give more explanation. This pattern will pass because of the logic in 
  DateTimeParseContext that won't raise an error when na value for the field is set twice but with the same value
```
int setParsedField(TemporalField field, long value, int errorPos, int successPos) {
        Objects.requireNonNull(field, "field");
        Long old = currentParsed().fieldValues.put(field, value);
        return (old != null && old.longValue() != value) ? ~errorPos : successPos;
    }
```